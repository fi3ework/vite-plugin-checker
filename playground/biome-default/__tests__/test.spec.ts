import stringify from 'fast-json-stable-stringify'
import fs from 'node:fs'
import { describe, expect, it } from 'vitest'
import {
  diagnostics,
  editFile,
  expectStderrContains,
  isBuild,
  isServe,
  stripedLog,
  resetDiagnostics,
  resetReceivedLog,
  sleepForEdit,
  sleepForServerReady,
} from '../../testUtils'
import path from 'node:path'

describe('biome', () => {
  describe.runIf(isServe)('serve', () => {
    it('get initial error and subsequent error', async () => {
      await sleepForServerReady()
      expect(stringify(diagnostics)).toMatchSnapshot()
      expect(stripedLog).toMatchSnapshot()

      console.log('-- edit error file --')
      resetReceivedLog()
      resetDiagnostics()
      editFile('src/index.js', (code) => code.replace(`var b = 'world'`, `const b = 'world'`))
      await sleepForEdit()
      expect(stringify(diagnostics)).toMatchSnapshot()
      expect(stripedLog).toMatchSnapshot()
    })
  })

  describe.runIf(isBuild)('build', () => {
    it('should fail', async () => {
      const expectedMsg = ['Use let or const instead of var', 'Found 2 errors']
      expectStderrContains(stripedLog, expectedMsg)
    })
  })
})
