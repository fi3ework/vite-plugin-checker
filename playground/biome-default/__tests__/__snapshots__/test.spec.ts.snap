// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`biome > serve > get initial error and subsequent error 1`] = `"[{"checkerId":"Biome","frame":"    1 | const a = 'hello'/n  > 2 | var b = 'world'/n      | ^^^^^^^^^^^^^^^/n    3 | const c = '!'/n    4 | var d = a + b + c/n    5 |","id":"<PROJECT_ROOT>/playground-temp/biome-default/src/index.js","level":1,"loc":{"column":1,"file":"<PROJECT_ROOT>/playground-temp/biome-default/src/index.js","line":2},"message":"[lint/style/noVar] Use let or const instead of var.","stack":""},{"checkerId":"Biome","frame":"    2 | var b = 'world'/n    3 | const c = '!'/n  > 4 | var d = a + b + c/n      | ^^^^^^^^^^^^^^^^^/n    5 |","id":"<PROJECT_ROOT>/playground-temp/biome-default/src/index.js","level":1,"loc":{"column":1,"file":"<PROJECT_ROOT>/playground-temp/biome-default/src/index.js","line":4},"message":"[lint/style/noVar] Use let or const instead of var.","stack":""}]"`;

exports[`biome > serve > get initial error and subsequent error 2`] = `
[
  " ERROR(Biome)  [lint/style/noVar] Use let or const instead of var.
 FILE  <PROJECT_ROOT>/playground-temp/biome-default/src/index.js:2:1

    1 | const a = 'hello'
  > 2 | var b = 'world'
      | ^^^^^^^^^^^^^^^
    3 | const c = '!'
    4 | var d = a + b + c
    5 |
",
  " ERROR(Biome)  [lint/style/noVar] Use let or const instead of var.
 FILE  <PROJECT_ROOT>/playground-temp/biome-default/src/index.js:4:1

    2 | var b = 'world'
    3 | const c = '!'
  > 4 | var d = a + b + c
      | ^^^^^^^^^^^^^^^^^
    5 |
",
  "[Biome] Found 2 errors and 0 warning",
]
`;

exports[`biome > serve > get initial error and subsequent error 3`] = `"[{"checkerId":"Biome","frame":"    2 | const b = 'world'/n    3 | const c = '!'/n  > 4 | var d = a + b + c/n      | ^^^^^^^^^^^^^^^^^/n    5 |","id":"<PROJECT_ROOT>/playground-temp/biome-default/src/index.js","level":1,"loc":{"column":1,"file":"<PROJECT_ROOT>/playground-temp/biome-default/src/index.js","line":4},"message":"[lint/style/noVar] Use let or const instead of var.","stack":""}]"`;

exports[`biome > serve > get initial error and subsequent error 4`] = `
[
  " ERROR(Biome)  [lint/style/noVar] Use let or const instead of var.
 FILE  <PROJECT_ROOT>/playground-temp/biome-default/src/index.js:4:1

    2 | const b = 'world'
    3 | const c = '!'
  > 4 | var d = a + b + c
      | ^^^^^^^^^^^^^^^^^
    5 |
",
  "[Biome] Found 1 error and 0 warning",
]
`;
